File: core/java/src/main/java/hara/lang/protocol/IDissoc.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.idissoc", name = "IDissoc")
public interface IDissoc<K> {
  @HaraMethod(value = "dissoc", arity = 2)
  IDissoc<K> dissoc(K k);
}

File: core/rust/src/lang/protocol/idissoc.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.idissoc", name = "IDissoc")]
pub trait IDissoc<K> {
    type Output;

    #[hara_method(value = "dissoc", arity = 2)]
    fn dissoc(&self, key: &K) -> Self::Output;
}
